common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/FederationBatchInfo.java

```diff
@@ -16,27 +16,33 @@
   private final String batchTag;
   private final LocalDate date;
   private FederationBatchStatus status;
+  private FederationBatchTarget targetSystem;
 
   /**
    * Creates a FederationBatchInfo and sets its status to {@link FederationBatchStatus#UNPROCESSED}.
-   * @param batchTag id of the batch
-   * @param date date the batch was created
+   *
+   * @param batchTag     id of the batch.
+   * @param date         date the batch was created.
+   * @param targetSystem the target system to use for download.
    */
-  public FederationBatchInfo(String batchTag, LocalDate date) {
-    this(batchTag, date, FederationBatchStatus.UNPROCESSED);
+  public FederationBatchInfo(String batchTag, LocalDate date, FederationBatchTarget targetSystem) {
+    this(batchTag, date, FederationBatchStatus.UNPROCESSED, targetSystem);
   }
 
   /**
    * Creates a FederationBatchInfo.
+   *
    * @param batchTag id of the batch
-   * @param date date the batch was created
-   * @param status status stored as {@link FederationBatchStatus}
+   * @param date     date the batch was created
+   * @param status   status stored as {@link FederationBatchStatus}
    */
   @PersistenceConstructor
-  public FederationBatchInfo(String batchTag, LocalDate date, FederationBatchStatus status) {
+  public FederationBatchInfo(String batchTag, LocalDate date, FederationBatchStatus status,
+      FederationBatchTarget targetSystem) {
     this.batchTag = batchTag;
     this.date = date;
     this.status = status;
+    this.targetSystem = targetSystem;
```

Comment:
I know we referred to the gateway as a target system during discussions, but in the case of Download it represents rather a "source" system